kubespawner/spawner.py

```diff
@@ -2105,18 +2105,27 @@ async def progress(self):
             return
 
         self.log.debug('progress generator: %s', self.pod_name)
-        start_future = self._start_future
+        start_future = None
```

Comment:
Leaving this first line is fine. The checks below help, but we don't need to wait to set it.

```suggestion
        start_future = self._start_future
```